src/snowflake/connector/result_set.py

```diff
@@ -164,13 +164,13 @@ def _fetch_arrow_batches(
         self._can_create_arrow_iter()
         return self._create_iter(iter_unit=IterUnit.TABLE_UNIT, structure="arrow")
 
-    def _fetch_arrow_all(self) -> Table | None:
+    def _fetch_arrow_all(self) -> Table:
```

Comment:
you also need to make the code change here
```suggestion
    def _fetch_arrow_all(self, force_return_table: bool = False) -> Table:
```